middleware/transaction/source/manager/handle.cpp

```diff
@@ -439,6 +381,45 @@ namespace casual
                      });
                   }
 
+                  template< typename Reply>
+                  void rollback_send_reply( State& state, const common::transaction::ID& origin, const Reply& reply, const detail::coordinate::Destination& destination)
+                  {
+                     Trace trace{ "transaction::manager::handle::local::detail::coordinate::rollback_send_reply"};
+
+                     // There should almost always be a a caller to the rollback/commit, but we might have started a rollback our selves.
+                     if( destination.process != common::process::handle())
+                     {
+                        common::log::line( verbose::log, "reply: ", reply);
+                        state.multiplex.send( destination.process.ipc, reply);
+                     }
+                     else
+                     {
+                        // We only start rollbacks our selves.
+                        CASUAL_ASSERT( ( concepts::any_of< Reply, common::message::transaction::rollback::Reply>));
```

Comment:
Why both the assert and constexpr if?

Reply:
The `constexpr if` is to only generate code for rollback-reply. Otherwise we get a compilation error, because it could be a commit-reply, even though we should never get to this branch with a commit-reply. Which the `assert` validates.